src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,17 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  const digits = [];
+
+  while (digits.length < 4) {
+    const randomDigit = Math.floor(Math.random() * 10);
+
+    if (!digits.includes(randomDigit) && randomDigit !== 0) {
```

Comment:
The condition `randomDigit !== 0` should be applied only when the `digits` array is empty to ensure the first digit is not 0. Consider separating the logic to handle the first digit differently.